### PR TITLE
use-double-click: add types

### PR DESCRIPTION
**Pull request by MasterOdin**
Adds a types file for [use-double-click](https://npmjs.com/package/use-double-click) package.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@MasterOdin thx!

**Review by martin-badin — APPROVED** [review]

File: types/use-double-click/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export default function useDoubleClick<T = unknown>(options: {
+  /** Dom node to watch for double clicks */
+  ref: RefObject<T> | MutableRefObject<T>;
+  /** The amount of time (in milliseconds) to wait before differentiating a single from a double click. Defaults to 300. */
+  latency?: number;
+  /** A callback function for single click events */
+  onSingleClick?: (e: MouseEvent) => void;
+  /** A callback function for double click events */
+  onDoubleClick?: (e: MouseEvent) => void;
+}): void;
```
Comment (DanielRosenwasser) [inline types/use-double-click/index.d.ts:17]: This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like in the following:

```ts
class Foo {
  // ...
}

export = Foo
```

To import the entity using a default import, you will need to enable `esModuleInterop`. Alternatively, you can just import using a more appropriate syntax like

```ts
import foo = require("foo");
```

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

File: types/use-double-click/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Type definitions for use-double-click 1.0
+// Project: https://github.com/tim-soft/use-double-click
+// Definitions by: Matthew Peveler <https://github.com/MasterOdin>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { RefObject, MutableRefObject, MouseEvent } from 'react';
+
+export default function useDoubleClick<T = unknown>(options: {
```
Comment (DanielRosenwasser) [inline types/use-double-click/index.d.ts:8]: This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like in the following:

```ts
class Foo {
  // ...
}

export = Foo
```

To import the entity using a default import, you will need to enable `esModuleInterop`. Alternatively, you can just import using a more appropriate syntax like

```ts
import foo = require("foo");
```

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/use-double-click/index.d.ts (hunk 2 above)
Comment (MasterOdin) [reply types/use-double-click/index.d.ts]: Done

**Review by peterblazejewicz — APPROVED** [review]

**Pull request merged**